packages/circuit-ui/components/Modal/Modal.tsx

```diff
@@ -13,192 +13,217 @@
  * limitations under the License.
  */
 
-import React, { FC, MouseEvent, KeyboardEvent, ReactNode } from 'react';
-import ReactModal, { Props } from 'react-modal';
-import { ClassNames } from '@emotion/core';
-import { useTheme } from 'emotion-theming';
+/** @jsx jsx */
+import { jsx, css, ClassNames } from '@emotion/core';
+import { ReactNode } from 'react';
+import ReactModal from 'react-modal';
 import { Theme } from '@sumup/design-tokens';
-import { Dispatch as TrackingProps } from '@sumup/collector';
-import noScroll from 'no-scroll';
 
-import IS_IOS from '../../util/ios';
 import { isFunction } from '../../util/type-check';
-import useClickHandler from '../../hooks/use-click-handler';
-
-type OnClose = (event?: MouseEvent | KeyboardEvent) => void;
-
-export interface ModalProps extends Partial<Props> {
-  children: ReactNode | (({ onClose }: { onClose?: OnClose }) => ReactNode);
+import useClickHandler from '../../hooks/useClickHandler';
+import { ModalComponent, BaseModalProps } from '../ModalContext/ModalContext';
+import CloseButton from '../CloseButton';
+
+const TRANSITION_DURATION_MOBILE = 120;
+const TRANSITION_DURATION_DESKTOP = 240;
+const TRANSITION_DURATION = Math.max(
+  TRANSITION_DURATION_MOBILE,
+  TRANSITION_DURATION_DESKTOP,
+);
+
+const closeButtonStyles = (theme: Theme) => css`
+  position: absolute;
+  top: ${theme.spacings.byte};
+  right: ${theme.spacings.byte};
+
+  ${theme.mq.kilo} {
+    top: ${theme.spacings.mega};
+    right: ${theme.spacings.mega};
+  }
+`;
+
+export interface ModalProps extends BaseModalProps {
   /**
-   * Determines if the modal is visible or not.
+   * TODO: Add description
    */
-  isOpen?: boolean;
+  children:
+    | ReactNode
+    | (({ onClose }: { onClose: BaseModalProps['onClose'] }) => ReactNode);
   /**
-   * Function to close the modal. Passed down to the children
-   * render prop.
+   * TODO: Add description
    */
-  onClose?: OnClose;
+  variant: 'immersive' | 'contextual';
```

Comment:
I don't remember the difference between these 🙈 the `contextual` variant doesn't seem to work as expected on Storybook (or does it?)

Reply:
It works if you reload the page and open the contextual one first (but then you can't open the immersive one). It's a bug that I still need to fix. A very, very weird bug.